cmake/modules/FindGMock.cmake

```diff
@@ -113,172 +117,329 @@ function(GTEST_ADD_TESTS executable extra_args)
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(GTest DEFAULT_MSG GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY)
+find_package_handle_standard_args(GMock DEFAULT_MSG GMOCK_LIBRARY GMOCK_INCLUDE_DIR GMOCK_MAIN_LIBRARY)
+
+include(CMakeFindDependencyMacro)
+find_dependency(Threads)
+
+set_target_properties(GTest::GTest PROPERTIES
+    INTERFACE_LINK_LIBRARIES "Threads::Threads"
+    IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
+    IMPORTED_LOCATION "${GTEST_LIBRARY}"
     )
 
-    add_library(libgtest IMPORTED STATIC GLOBAL)
-    add_dependencies(libgtest gmock)
-    add_library(libgmock IMPORTED STATIC GLOBAL)
-    add_dependencies(libgmock gmock)
-    
-    ExternalProject_Get_Property(gmock source_dir binary_dir)
-    set(GTEST_INCLUDE_DIR  "${source_dir}/gtest/include")
-    mark_as_advanced(GTEST_INCLUDE_DIR)
-    set(GTEST_LIBRARY      "${binary_dir}/gtest/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gtest${Suffix}")
-    mark_as_advanced(GTEST_LIBRARY)
-    set(GTEST_MAIN_LIBRARY "${binary_dir}/gtest/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gtest_main${Suffix}")
-    mark_as_advanced(GTEST_MAIN_LIBRARY)
-    set(GMOCK_INCLUDE_DIR  "${source_dir}/include")
-    mark_as_advanced(GMOCK_INCLUDE_DIR)
-    set(GMOCK_LIBRARY      "${binary_dir}/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gmock${Suffix}")
-    mark_as_advanced(GMOCK_LIBRARY)
-    set(GMOCK_MAIN_LIBRARY "${binary_dir}/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gmock_main${Suffix}")
-    mark_as_advanced(GMOCK_MAIN_LIBRARY)
+if(GTEST_INCLUDE_DIRS)
+    set_target_properties(GTest::GTest PROPERTIES
+        INTERFACE_INCLUDE_DIRECTORIES "${GTEST_INCLUDE_DIRS}")
 endif()
 
-set_target_properties(libgtest PROPERTIES
-    "IMPORTED_LOCATION" "${GTEST_LIBRARY}"
-    "IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}"
-    "INTERFACE_INCLUDE_DIRECTORIES" "${GTEST_INCLUDE_DIR}"
-)
-set_target_properties(libgmock PROPERTIES
-    "IMPORTED_LOCATION" "${GMOCK_LIBARY}"
-    "IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}"
-    "INTERFACE_INCLUDE_DIRECTORIES" "${GMOCK_INCLUDE_DIR}"
-)
+set_target_properties(GTest::Main PROPERTIES
+    INTERFACE_LINK_LIBRARIES "GTest::GTest"
+    IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
+    IMPORTED_LOCATION "${GTEST_MAIN_LIBRARY}")
 
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(GTest DEFAULT_MSG GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(GMock DEFAULT_MSG GMOCK_LIBRARY GMOCK_INCLUDE_DIR GMOCK_MAIN_LIBRARY)
+set_target_properties(GMock::GMock PROPERTIES
+    INTERFACE_LINK_LIBRARIES "Threads::Threads"
+    IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
+    IMPORTED_LOCATION "${GMOCK_LIBRARY}")
+
+if(GMOCK_INCLUDE_DIRS)
+    set_target_properties(GMock::GMock PROPERTIES
+        INTERFACE_INCLUDE_DIRECTORIES "${GMOCK_INCLUDE_DIRS}")
+endif()
+
+set_target_properties(GMock::Main PROPERTIES
+    INTERFACE_LINK_LIBRARIES "GMock::GMock"
```

Comment:
This is how it's done in a recent FindGTest.cmake provided with CMake. That's why I didn't consider removing it.